src/MultiAgentReinforcementLearning/src/MultiAgentReinforcementLearning.jl

```diff
@@ -0,0 +1,224 @@
+module MultiAgentReinforcementLearning
+
+export MultiAgentPolicy
+export MultiAgentHook
+
+using ReinforcementLearningBase
+using ReinforcementLearningCore
+
+import ReinforcementLearningCore: RLCore
+import Base.getindex
+import Base.iterate
+
+struct MultiAgentPolicy <: AbstractPolicy
+    agents::NamedTuple
```

Comment:
This is going to be dynamically dispatched as `NamedTuple` is not a concrete type.
```suggestion
struct MultiAgentPolicy{NT<: NamedTuple} <: AbstractPolicy
    agents::NT
```